core/src/main/scala/org/apache/spark/ExecutorAllocationClient.scala

```diff
@@ -59,6 +59,19 @@ private[spark] trait ExecutorAllocationClient {
   def killExecutors(executorIds: Seq[String]): Seq[String]
 
   /**
+   * Request that the cluster manager try harder to kill the specified executors,
+   * and maybe replace them.
+   * @return whether the request is acknowledged by the cluster manager.
+   */
+  def killExecutors(executorIds: Seq[String], replace: Boolean, force: Boolean): Seq[String]
```

Comment:
I wonder if it's not better to just change the existing method to have the extra arguments with default values. I see only a handful of calls to the 1-arg version, so maybe even forcing all of them to pass all arguments would be ok.